packages/router/src/RouterLink.ts

```diff
@@ -93,7 +95,40 @@ export function useLink(props: UseLinkOptions) {
   const router = inject(routerKey)!
   const currentRoute = inject(routeLocationKey)!
 
-  const route = computed(() => router.resolve(unref(props.to)))
+  const isValidTo = (to: unknown) => typeof to === 'string' || isObject(to)
+  let hasPrevious = false
+  let previousTo: unknown = null
```

Comment:
I think these can be merged in one

Reply:
I tried that when I first wrote it, but it didn't work out.

The problem is picking an initial value that can be used as a sentinel. `null` and `undefined` don't work, because those could be set later.

I tried using a Symbol or object instead, something like this:

```js
const INITIAL_VALUE = {}
let previousTo: unknown = INITIAL_VALUE
```

with subsequent checks for `previousTo === INITIAL_VALUE`, but that seemed more complicated than a separate `hasPrevious` flag.